src/main/java/com/launchdarkly/client/FeatureFlag.java

```diff
@@ -0,0 +1,242 @@
+      visited.add(key);
+      if (visited.contains(prereq.getKey())) {
+        logger.error("Prerequisite cycle detected when evaluating feature flag: " + key);
+        return null;
+      }
+      FeatureFlag prereqFeatureFlag = featureStore.get(prereq.getKey());
+      if (prereqFeatureFlag == null) {
+        logger.error("Could not retrieve prerequisite flag: " + prereq.getKey() + " when evaluating: " + key);
+        return null;
+      }
+      JsonElement prereqValue;
+      if (prereqFeatureFlag.isOn()) {
+        EvalResult prereqEvalResult = prereqFeatureFlag.evaluate(user, featureStore, events, visited);
+        if (prereqEvalResult == null) {
+          return null;
+        }
+        prereqValue = prereqEvalResult.value;
+        visited = prereqEvalResult.visitedFeatureKeys;
+        events = prereqEvalResult.prerequisiteEvents;
+        events.add(new FeatureRequestEvent(prereqFeatureFlag.getKey(), user, prereqValue, null));
+        if (prereqValue == null || !prereqValue.equals(prereqFeatureFlag.getVariation(prereq.getVariation()))) {
+          return new EvalResult(null, events, visited);
+        }
+      } else {
+        return null;
+      }
+    }
+    return new EvalResult(getVariation(evaluateIndex(user)), events, visited);
+  }
+
+  private Integer evaluateIndex(LDUser user) {
+    // Check to see if targets match
+    for (Target target : targets) {
+      for (String v : target.getValues()) {
+        if (v.equals(user.getKey().getAsString())) {
+          return target.getVariation();
+        }
+      }
+    }
+
+    // Now walk through the rules and see if any match
+    for (Rule rule : rules) {
+      if (rule.matchesUser(user)) {
+        return rule.variationIndexForUser(user, key, salt);
+      }
+    }
+
+    // Walk through the fallthrough and see if it matches
+    return fallthrough.variationIndexForUser(user, key, salt);
+  }
+
+  private JsonElement getVariation(Integer index) {
+    if (index == null || index >= variations.size()) {
+      return null;
+    } else {
+      return variations.get(index);
+    }
+  }
+
+  public int getVersion() {
+    return version;
+  }
+
+  void setVersion(int version) {
+    this.version = version;
+  }
+
+  public String getKey() {
+    return key;
+  }
+
+  public boolean isDeleted() {
+    return deleted;
+  }
+
+  public void setDeleted() {
+    deleted = true;
+  }
+
+  public boolean isOn() {
+    return on;
+  }
+
+  public List<Prerequisite> getPrerequisites() {
+    return prerequisites;
+  }
+
+  public String getSalt() {
+    return salt;
+  }
+
+  public String getSel() {
+    return sel;
+  }
+
+  public List<Target> getTargets() {
+    return targets;
+  }
+
+  public List<Rule> getRules() {
+    return rules;
+  }
+
+  public Rule getFallthrough() {
+    return fallthrough;
+  }
+
+  public List<JsonElement> getVariations() {
+    return variations;
+  }
+
+  @Override
+  public boolean equals(Object o) {
```

Comment:
I'm not sure there's a need to maintain auto-generated equals and hashcode implementations.